Erfan

erfannayeb opened the pull request.

erfannayeb committed:
Update Board.java

aaa

erfannayeb committed:
Update Board.java

new aaa

erfannayeb committed:
Update Bead.java

bead

Adjective10111 commented on Quoridor/src/game/Board.java:
```diff
@@ -1,4 +1,41 @@
+        bead2.setX(8);
+        bead2.setY(4);
+    }
+        public void move(String playermove){
+            if(playermove.equalsIgnoreCase("Up")) {
+                board[bead1.getX()][bead1.getY()]=false;
+                bead1.setX(bead1.getX()-1);
+                board[bead1.getX()][bead1.getY()]=true;
+            }
+
+            if(playermove.equalsIgnoreCase("Down")){
+                board[bead1.getX()][bead1.getY()]=false;
+                bead1.setX(bead1.getX()+1);
+                board[bead1.getX()][bead1.getY()]=true;
+            }
+            if (playermove.equalsIgnoreCase("Right")){
+                board[bead1.getX()][bead1.getY()]=false;
+                bead1.setY(bead1.getY()+1);
+                board[bead1.getX()][bead1.getY()]=true;
+            }
+            if (playermove.equalsIgnoreCase("Left")){
+                board[bead1.getX()][bead1.getY()]=false;
+                bead1.setY(bead1.getY()-1);
+                board[bead1.getX()][bead1.getY()]=true;
+            }
+        }
+
+    }
```
کل این متد اشکال داره چون فقط یه مهره رو تکون میده دوتا کار انجام بده
- make Player class with these attributes
```
Bead bead = new Bead();
Wall[] walls = new Wall[8];
```
- make turn method
  - add this attribute to Board
     ```
     Player turn = player1;
     ```
  - add this method
     ```
     turn() {
           if (this.turn == player1)
                 turn = player2;
           else
                 turn = player1;
     }
     ```
بعد بجای دوتا بید دوتا پلیر اضاف کن و آخر متد موو از متد ترن استفاده کن و بجای
bead1
بزار:
turn.bead

Ehsanshr replied:
صفت به نظر من نیازی به کلاس پلیر نیس
و چرا کلاس وال رو اونطوری ساختی؟؟! چی تو ذهنته؟؟
نمیشه که هر یه دیوارو جدا بزاریم ک
اینطوری چه اتریبیوت ها و چه متد هایی میخوای تو کلاس وال بزاری؟
طبق چیزی ک تو میگی یا علاوه بر کلاس وال باید یه کلاس والز هم داشته باشیم که همه دیوارارو در بر بگیره 
یا باید توی بورد همه دیوارا و اینکه بسته شدن یا باز موندن راه برای مهره رو بررسی کنیم که ایا میتونه برنده بشه یا نه
که به نظرم نیازی نیس اصلا و میتونیم توی کلاس وال همه دیوارا رو مورد بررسی قرار بدیم و دوتا بید رو به عنوان پارامتر براش بفرستیم برا اینکه بررسی کنیم اگه دیواری قراربگیره زندانی نشه و دو تا متغیر که تعداد دیوارای باقی مونده هر مهره رو نشون بده که اولش برا هر کدوم ۸عه حالا بخش وال رو تا یه جاییش زدم اوکی شد میفرستمش

Adjective10111 replied:
> صفت به نظر من نیازی به کلاس پلیر نیس
> و چرا کلاس وال رو اونطوری ساختی؟؟! چی تو ذهنته؟؟
> نمیشه که هر یه دیوارو جدا بزاریم ک
> اینطوری چه اتریبیوت ها و چه متد هایی میخوای تو کلاس وال بزاری؟
> طبق چیزی ک تو میگی یا علاوه بر کلاس وال باید یه کلاس والز هم داشته باشیم که همه دیوارارو در بر بگیره 
> یا باید توی بورد همه دیوارا و اینکه بسته شدن یا باز موندن راه برای مهره رو بررسی کنیم که ایا میتونه برنده بشه یا نه
> که به نظرم نیازی نیس اصلا و میتونیم توی کلاس وال همه دیوارا رو مورد بررسی قرار بدیم و دوتا بید رو به عنوان پارامتر براش بفرستیم برا اینکه بررسی کنیم اگه دیواری قراربگیره زندانی نشه و دو تا متغیر که تعداد دیوارای باقی مونده هر مهره رو نشون بده که اولش برا هر کدوم ۸عه حالا بخش وال رو تا یه جاییش زدم اوکی شد میفرستمش

باشه بزن فعلا تا ببینم چی میگی چون نفهمیدم😅✋🏼

erfannayeb replied:
baray man ok bud?agar ok bud kare badi k bayad anjam bedam bgid

Adjective10111 replied:
لازم نبود turn.bead تو پرانتز بزاریا
ولی درکل اوکیه:+1:

erfannayeb replied:
are midonam injori rahat budi

Adjective10111 reviewed (commented):
Check it out

erfannayeb committed:
Create Player

erfannayeb committed:
Delete Player

erfannayeb committed:
Add files via upload

erfannayeb committed:
Update Board.java

erfannayeb committed:
new

Adjective10111 committed:
Little Tweaks

In the ifs you always asked for `getX` which was **wrong** (some times the `getY` should have been checked)
You *didn't fill* the array with 'e' which is fine I did it for you
And also you used the *x, y* like this: `[x][y]` , but initially you wrote `board[0][8] = 'B'` which is `[y][x]` and does not match
And finally I made it a little more spaced and easy to read

Adjective10111 reviewed (commented):
In the ifs you always asked for `getX` which was **wrong** (some times the `getY` should have been checked)
You *didn't fill* the array with 'e' which is fine I did it for you
And also you used the *x, y* like this: `[x][y]` , but initially you wrote `board[0][8] = 'B'` which is `[y][x]` and does not match
And finally I made it a little more spaced and easy to read

erfannayeb committed:
some changes

Adjective10111 reviewed (commented):
- Don't remove spaces😑
- I will change it a little don't mind me😁

Adjective10111 committed:
Update Board.java

erfannayeb commented:
khat 8-9 error mide
fekr konam hamon setx getx ina bayad bezanam

erfannayeb committed:
Update Board.java

erfannayeb committed:
Update Board.java

erfannayeb committed:
Update Player.java

erfannayeb committed:
Add files via upload

erfannayeb committed:
win method

erfannayeb committed:
Update Player.java

erfannayeb committed:
Update Bead.java

erfannayeb committed:
Update Controller.java

erfannayeb committed:
Update Play.java

erfannayeb committed:
Add files via upload

erfannayeb committed:
Add files via upload

erfannayeb committed:
Add files via upload

erfannayeb committed:
Delete sample.fxml

erfannayeb committed:
Update Controller.java

changes

erfannayeb committed:
changes

erfannayeb committed:
Update Board.java

erfannayeb committed:
Update Controller.java
